integration_test/cloudbuild-integration-test.yaml

```diff
@@ -1,602 +1,52 @@
-
-# example-redacted has replaced UIDs and removed ContentTime,ContentDate tags
-- name: 'busybox:latest'
-  id: check-store-diff-destination-2
-  args:
-  - diff
-  - 'integration_test/downloaded-destination-2.dcm'
-  - 'integration_test/data/example-redacted-mg-jp2k.dcm'
-  waitFor:
-  - check-store-curl-destination-2
-
-- name: 'busybox:latest'
-  id: check-store-diff-with-local-backup
-  args:
-    - diff
-    - 'integration_test/downloaded2.dcm'
-    - 'integration_test/data/example2.dcm'
-  waitFor:
-    - check-store-curl-with-local-backup
-
-- name: 'busybox:latest'
-  id: check-store-diff-with-gcs-backup
-  args:
-    - diff
-    - 'integration_test/downloaded3.dcm'
-    - 'integration_test/data/example3.dcm'
-  waitFor:
-    - check-store-curl-with-gcs-backup
-
-- name: 'busybox:latest'
-  id: check-find-diff-instance
-  args:
-  - diff
-  - 'integration_test/findscu-instance-result1.xml'
-  - 'integration_test/data/findscu-instance-expected.xml'
-  waitFor:
-  - run-find-scu-instance
-
-- name: 'busybox:latest'
-  id: check-find-diff-series
-  args:
-  - diff
-  - 'integration_test/findscu-series-result1.xml'
-  - 'integration_test/data/findscu-series-expected.xml'
-  waitFor:
-  - run-find-scu-series
-
-- name: 'busybox:latest'
-  id: check-find-diff-study
-  args:
-  - diff
-  - 'integration_test/findscu-study-result1.xml'
-  - 'integration_test/data/findscu-study-expected.xml'
-  waitFor:
-  - run-find-scu-study
-
-# storescp adds 1 extra tag
-# (0002,0016) [AE] SourceApplicationEntityTitle: IMPORTADAPTER
-- name: 'busybox:latest'
-  id: check-move-diff
-  args:
-  - diff
-  - 'integration_test/storescp-data/${_REPLACED_UID}'
-  - 'integration_test/data/example-redacted-moved-jp2k.dcm'
-  waitFor:
-  - run-move-scu
-
-- name: ${_MAVEN_IMAGE}
-  id: check-commitment-diff
-  volumes:
-  - name: 'vol1'
-    path: '/root/.m2'
-  args:
-  - bash
-  - /workspace/integration_test/scripts/diff-commitment-result.sh
-  dir: "dcm4che/dcm4che-tool/dcm4che-tool-dcm2xml"
-  waitFor:
-  - run-commitment-scu
-
-- name: ${_CLOUD_SDK_IMAGE}
-  id: delete-dicom-store
-  args:
-  - gcloud
-  - beta
-  - healthcare
-  - dicom-stores
-  - delete
-  - ${_STORE_NAME}
-  - "--project=${_PROJECT}"
-  - "--dataset=${_DATASET}"
-  - "--location=${_LOCATION}"
-  - "--quiet"
-  waitFor:
-  - check-store-curl
-  - check-store-curl-destination-2
-  - check-store-curl-with-local-backup
-  - check-store-curl-with-gcs-backup
-  - run-find-scu-instance
-  - run-find-scu-series
-  - run-find-scu-study
-  - run-move-scu
-  - run-commitment-scu
-
-- name: ${_CLOUD_SDK_IMAGE}
-  id: delete-dicom-store-destination-2
-  args:
-  - gcloud
-  - beta
-  - healthcare
-  - dicom-stores
-  - delete
-  - "${_STORE_NAME}-destination-2"
-  - "--project=${_PROJECT}"
-  - "--dataset=${_DATASET}"
-  - "--location=${_LOCATION}"
-  - "--quiet"
-  waitFor:
-  - delete-dicom-store
+    - python
+    - ./integration_test/step_4.py
 
+# publish image
 - name: ${_CLOUD_SDK_IMAGE}
```

Comment:
I'm not sure if publish will be able to pick up the image built in step 1, but I could submit this PR after the other comment is addressed and try. If it does not work, we can roll back and try to fix the issue.

Reply:
This is part of the logs obtained by adding the docker images | grep "healthcare-api-dicom-dicomweb-adapter-" to publish-image.sh  script:
_Step #4 - "publish-image": gcr.io/cloud-healthcare-containers/healthcare-api-dicom-dicomweb-adapter-export   0.2.24                           3506cea85431        3 minutes ago       715MB
Step #4 - "publish-image": gcr.io/cloud-healthcare-containers/healthcare-api-dicom-dicomweb-adapter-export   local_run                        3506cea85431        3 minutes ago       715MB
Step #4 - "publish-image": gcr.io/cloud-healthcare-containers/healthcare-api-dicom-dicomweb-adapter-import   0.2.24                           9b28e305c040        4 minutes ago       721MB
Step #4 - "publish-image": gcr.io/cloud-healthcare-containers/healthcare-api-dicom-dicomweb-adapter-import   local_run                        9b28e305c040        4 minutes ago       721MB
Step #4 - "publish-image": ERROR: (gcloud.kms.decrypt) NOT_FOUND: CryptoKey projects/dicom-adapter-261020/locations/global/keyRings/default/cryptoKeys/github-robot-access-token not found.
Finished Step #4 - "publish-image"
...
cd7bc1ff-38a3-4585-859f-cdc6a7d1b3ee  2021-02-05T10:21:19+00:00  10M53S    gs://dicom-adapter-261020_cloudbuild/source/1612520469.519869-3a410f4fe61643c3b6e4e9946ef8dc6b.tgz  -       SUCCESS_

We can see that the images are present and we can use them to publish. I receive an error related to the lack of required credentials for publishing. It is possible that the logs will differ for updating the PR, but I do not have the permission to view the details of the test passing logs

Reply:
Thanks, seems like we have some confidence in this working then.